packages/docs-site/blog/2023/06/29/new-language-features.md

```diff
@@ -0,0 +1,302 @@
+
+(For concrete examples see Professor Keenan Crane's [GitHub issue](https://github.com/penrose/penrose/issues/785)).
+
+### What have we done?
+
+We [added](https://github.com/penrose/penrose/pull/1074) two reserved variables in each Style block:
+
+- `match_total` records how many times, in total, the Style block is matched, and
+- `match_id` records the `1`-indexed ordinal of the current matching.
+
+As an example,
+
+```
+-- dsl
+type MyType
+
+-- sub
+MyType T1, T2, T3, T4
+
+-- sty, canvas specifications omitted
+forall MyType t {
+  // can use `match_id` and `match_total`
+}
+```
+
+Since the Style block is matched four times, `match_total` equals to 4 within the Style block. Furthermore, `match_id` is one of 1, 2, 3, or 4, depending on which matching it represents currently.
+
+### How have we done it?
+
+For each Style block, we _augment_ the Style block body with two artificial AST nodes, as if they were parts of the original Style program:
+
+- One node visualizes to `match_total = <the total number of matches>`, and
+- The other visualizes to `match_id = <the current ordinal of the matching>`.
+
+Then we process the Style block as usual. Notice that now, `match_total` and `match_id` are assigned to the correct values and can be used later in the Style block as local variables.
+
+## Inline Comparison Operators
+
+We add syntactic sugars of `<`, `==`, and `>` to correspond to function calls `lessThan`, `equal`, and `greaterThan`. This makes Style-writing more convenient as it allows the users to write, for example,
+
+```
+ensure circle1.r < circle2.r
+```
+
+and have it be treated as equivalent to
+
+```
+ensure lessThan(circle1.r, circle2.r)
+```
+
+### How have we done it?
+
+We modified the grammar for constraint and objective declarations with
+
+```
+<constr> ::= ensure <body> <staged layout>+
+   <obj> ::= encourage <body> <staged layout>+
+
+  <body> ::= <identifier> ( <expr_list> )  // Function Call
+          |  <expr> <op> <expr>            // Inline Comparison
+
+    <op> ::= ==                            // Equal
+          |  <                             // Less Than
+          |  >                             // Greater Than
+```
+
+In other words, the `<body>` of a constraint or objective now can either be a "Function Call" or an "Inline Comparison."
+
+In the Style compiler where we handle constraints and objectives, we proceed as before when we encounter a "Function Call." When we encounter an "Inline Comparison," on the other hand, we handle it as if it is a "Function Call," with the corresponding function name (`lessThan`, `greaterThan`, or `equal`) and the two operands as parameters to the function.
+
+### Caveats
+
+Originally we wanted to implement operators `<=` and `>=`. However, there are no corresponding functions to these operators in the Penrose system. From an optimizer's perspective, in fact, `<=` and `<` are treated the same, and `>=` and `>` are treated the same.
```

Comment:
Add a conclusion section that summarizes and ends with some final thoughts.

Reply:
+1. Given the length of this post, having a tldr up top might also be good.